pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -595,6 +595,7 @@ private CompletableFuture<Long> individualAckNormal(CommandAck ack, Map<String,
                 ackedCount = getAckedCountForMsgIdNoAckSets(batchSize, position, ackOwnerConsumer);
                 if (checkCanRemovePendingAcksAndHandle(ackOwnerConsumer, position, msgId)) {
                     addAndGetUnAckedMsgs(ackOwnerConsumer, -(int) ackedCount);
+                    updateBlockedConsumerOnUnackedMsgs(ackOwnerConsumer);
```

Comment:
acks get modified also on line 592 in a similar way as on line 597. Just wondering if the logic in `updateBlockedConsumerOnUnackedMsgs` should be moved directly into `addAndGetUnAckedMsgs` method so that all changes would get covered.